### PR TITLE
bpo-33709: test_ntpath and test_posixpath fail in Windows with ACP!=1252

### PR DESCRIPTION
**Pull request by native-api**
https://bugs.python.org/issue33709

### CODE REVIEW

File: Lib/test/support/__init__.py [hunk 1]
```diff
@@ -844,7 +844,9 @@ def dec(*args, **kwargs):
     '\u20AC',
 ):
     try:
-        os.fsdecode(os.fsencode(character))
+        if os.fsdecode(os.fsencode(character))\
```
Comment (serhiy-storchaka) [inline Lib/test/support/__init__.py]: Why does this condition take two lines?

File: Lib/test/support/__init__.py [hunk 2]
```diff
@@ -844,7 +844,9 @@ def dec(*args, **kwargs):
...
+            != character:
+            raise UnicodeError()
```
Comment (serhiy-storchaka) [inline Lib/test/support/__init__.py]: Just `raise UnicodeError`.

File: Lib/test/support/__init__.py (hunk 1 above)
Comment (native-api) [reply Lib/test/support/__init__.py]: I rebased it from 2.7 where the expression is longer. No prob, will reformat.

File: Lib/test/support/__init__.py (hunk 2 above)
Comment (native-api) [reply Lib/test/support/__init__.py]: Ditto.

**Review by serhiy-storchaka — APPROVED** [review]

**Comment by native-api** [comment]
https://github.com/python/cpython/pull/7279 is backport to 2.7 (bedevere didn't detect it 'cuz it was added earlier)

**Comment by native-api** [comment]
Travis failure caused by https://bugs.python.org/issue33532

**Review by vstinner — DISMISSED** [review]
Sorry, I don't have the bandwidth to test properly your change on Windows right now, but I don't think that it's correct.


Python uses UTF-8 to encode paths on Windows since Python 3.6: https://vstinner.github.io/python36-utf8-windows.html It should not use the ANSI code page for filenames anymore (except if you explicitly opt-in for the ANSI code page).

**Comment by native-api** [comment]
@zooba Already did that in https://github.com/python/cpython/pull/7278/commits/9c9db384c65c950e725e45c0fd0500b6443855cc .

**Comment by serhiy-storchaka** [comment]
It looks to me that all comments were addressed. I'm going to merge this PR.

**Pull request merged**